content/news/20.4networking-rework.md

```diff
@@ -0,0 +1,224 @@
+---
+title: "The Networking Refactor"
+date: 2023-12-31T21:00:00+00:00
+publishDate: 2023-12-31T21:00:00+00:00
+categories:
+- News
+author: oriononline
+summary: |
+    This post gives an overview of the changes made to the networking system of NeoForge 20.4.
+description: |
+    This post gives an overview of the changes made to the networking system of NeoForge 20.4.
+---
+
+# Introduction
+## Reworked networking
+Welcome to the reworked networking blog post for NeoForged.
+The post will describe the changes to NeoForge to enable the configuration network protocol designed by Modmuss50, a maintainer of Fabric, with some minor modifications. I'm sharing more on those below.
+
+## SimpleChannel and EventChannel
+Forge contained two different ways of implementing a custom network channel. A simple registration-based approach is known as SimpleChannel, and a system that fires events for each packet received is known as EventChannel.
+To simplify using networking and make interacting with new packet layouts easier, it was decided to rework both implementations into a single system, combining the best of both worlds.
+
+# Implementation
+## New network payload handling
+The system is based on the `CustomPacketPayload` definition that Mojang uses to represent the content of custom packets. Internally, they register their custom payloads, used mainly for debugging, to a map. We need to extend this map so that a modder can send and receive a custom implementation of this `CustomPacketPayload`. This is what a significant part of this PR achieves. Modders can introduce new CustomPacketPayload implementations by registering them during the `RegisterPacketHandlerEvent`.
```

Comment:
```suggestion
The system is based on the `CustomPacketPayload` definition that Mojang uses to represent the content of custom packets. Internally, they register their custom payloads, used mainly for debugging, to a map. We need to extend this map so that a modder can send and receive a custom implementation of this `CustomPacketPayload`. This is what a significant part of the rework achieves. Modders can introduce new CustomPacketPayload implementations by registering them during the `RegisterPacketHandlerEvent`.
```